scripts/verify.ts

```diff
@@ -0,0 +1,93 @@
+import { promises as fs } from 'node:fs';
+import path from 'node:path';
+import util from 'node:util';
+
+import { ethers } from 'ethers';
+
+import type { SingletonDeploymentJSON } from '../src/types';
+
+type Options = {
+  version: string;
+  chainId: string;
+  rpc: string;
+  verbose: boolean;
+};
+
+function parseOptions(): Options {
+  const options = {
+    version: { type: 'string' },
+    chainId: { type: 'string' },
+    rpc: { type: 'string' },
+    verbose: { type: 'boolean' },
+  } as const;
+  const { values } = util.parseArgs({ options });
+
+  for (const option of ['version', 'chainId', 'rpc'] as const) {
+    if (values[option] === undefined) {
+      throw new Error(`missing --${option} flag`);
+    }
+  }
+
+  return {
+    ...(values as Options),
+    verbose: values.verbose === true,
+  };
+}
+
+async function main() {
+  const options = parseOptions();
+  const debug = (...msg: unknown[]) => {
```

Comment:
Doubt: Can the type be `string[]`?

Reply:
It can also take other JavaScript types: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Data_types

Reply:
Not in this case. In particular, this is the type of `console.log` style functions ([source](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/47c5b50ec30234ae2ee0c9ff6cdb658604941dbe/types/node/console.d.ts#L227))

```ts
log(message?: any, ...optionalParams: any[]): void;
```

Meaning that it accepts any `message` or optional parameters. We further narrow the type to `unknown` to prevent misuse of it in our `debug` function implementation.

Why? So we can do stuff like:
```ts
debug({ someObject: "with fields" });
```